kvdb/sqlite/db.go

```diff
@@ -0,0 +1,85 @@
+//go:build kvdb_sqlite
+// +build kvdb_sqlite
+
+package sqlite
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/lightningnetwork/lnd/kvdb/common_sql"
+	_ "modernc.org/sqlite" // Register relevant drivers.
+)
+
+const (
+	// sqliteOptionPrefix is the string prefix sqlite uses to set various
+	// options. This is used in the following format:
+	//   * sqliteOptionPrefix || option_name = option_value.
+	sqliteOptionPrefix = "_pragma"
+
+	// sqliteTxLockImmediate is dsn option used to ensure that write
+	// transactions are started immediately.
+	sqliteTxLockImmediate = "_txlock=immediate"
+)
+
+// postgresReplacements define a set of postgres keywords that should be swapped
+// out with certain other sqlite keywords in any queries.
+var postgresReplacements = common_sql.PostgresCmdReplacements{
+	"BYTEA":                 "BLOB",
```

Comment:
See the other comment about potentially reversing this operation (also name inconsistency here).